### PR TITLE
Feat/send user operation to tips

### PR DESCRIPTION
**Pull request by rayyan224**
Send user operation to TIPS

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td><code>2</code> if repo is sensitive</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>1</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>2</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

**Review by danyalprout — CHANGES_REQUESTED** [review]
I would remove the `TipsClient` completely and just use a Provider. Adding the `TipsClient` makes this harder to unit test (as there's no interface) and just duplicates the logic.

File: crates/account-abstraction/src/tips_client.rs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    #[test]
+    fn test_client_creation() {
+        let url = Url::parse("http://localhost:8080").unwrap();
+        let _client = TipsClient::new(url);
+    }
+
+    // Integration test - requires running tips service
+    #[tokio::test]
+    #[ignore]
+    async fn test_send_user_operation() -> Result<()> {
```
Comment (danyalprout) [inline crates/account-abstraction/src/tips_client.rs:111]: nit: I wouldn't check in tests that don't run as part of the CI
